### PR TITLE
Create reproducible non unique files when muxing or encoding

### PR DESCRIPTION
**Pull request by Nikoli**
Also prevents adding Unique ID and ffmpeg/libav version to files.

P.S. It seems libav has bug:
https://bugzilla.libav.org/show_bug.cgi?id=845

### CODE REVIEW

**Comment by Nikoli** [comment]
Since ```--format='bestvideo+bestaudio/best'``` is default this patch should mostly fix issue #2885.

File: youtube_dl/postprocessor/ffmpeg.py [hunk 1]
```diff
@@ -233,7 +233,7 @@ def run(self, information):
                 # MP3 otherwise.
                 acodec = 'libmp3lame'
                 extension = 'mp3'
-                more_opts = []
+                more_opts = ['-flags', '+bitexact',]
```
Comment (jaimeMF) [inline youtube_dl/postprocessor/ffmpeg.py]: Aren't you adding the same args in `run_ffmpeg`?:
```python
opts = ['-vn', '-fflags', '+bitexact'] + acodec_opts + more_opts
```

**Comment by Nikoli** [comment]
@jaimeMF, ```-fflags +bitexact``` affects only muxer and ```-flags +bitexact``` affects only encoder, so all ffmpeg/avconv commands should have ```-fflags +bitexact```. If encoding is done, then also ```-flags +bitexact`` should be added.

https://libav.org/avconv.html
https://git.libav.org/?p=libav.git;a=blob;f=tests/fate-run.sh;h=a0a69ad0a7ca716e008446eea67f69b19fb14840;hb=HEAD#l122

https://www.ffmpeg.org/ffmpeg-formats.html#toc-Format-Options
https://www.ffmpeg.org/ffmpeg-codecs.html#toc-Codec-Options
http://git.videolan.org/?p=ffmpeg.git;a=blob;f=tests/fate-run.sh;h=1a8ff940bd92f4e301435c49d0dc9331bda61700;hb=HEAD#l143

**Comment by Nikoli** [comment]
Simpler patch, now hls downloads are not unique too.

**Pull request closed without merging**